### ISSUE TITLE
Fetch is not working for android device in React native 0.62.2 even with HTTPS

### ISSUE THREAD

**Issue by SUMITIOS** [facebook/react-native#30012]
## Description
The api calls using `fetch` are not working for android only. The same api call using `XMLHttpRequests` is working well.

## React Native version:
System:
    OS: macOS 10.15.5
    CPU: (4) x64 Intel(R) Core(TM) i5-4260U CPU @ 1.40GHz
    Memory: 28.05 MB / 4.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.18.3 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.6 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.5, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK:
      API Levels: 23, 24, 25, 26, 27, 28, 29
      Build Tools: 23.0.1, 23.0.3, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.3, 28.0.3, 29.0.2, 29.0.3
      Android NDK: Not Found
  IDEs:
    Android Studio: 3.4 AI-183.5429.30.34.5452501
    Xcode: 11.5/11E608c - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_212 - /usr/bin/javac
    Python: 2.7.15 - /usr/local/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0 
    react-native: 0.62.2 => 0.62.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

## Steps To Reproduce

1. Create a fresh project and using react native cli.
2. call any api using fetch (http or https).
3. You can try to solve using android:usesCleartextTraffic="true" but it will not work.

## Expected Results
The api should work and give some response in android.

## Snack, code example, screenshot, or link to a repository:
` fetch('https://graph.facebook.com/v2.5/me?fields=email,name,friends&access_token=' + token)
            .then((response) => response.json())
            .then((json) => {
                console.warn(json)
            })
            .catch((err) => {
                console.warn(err.message)
            })`

**Comment by hsnandeeshgowda** [comment]
what is it consoling ?
check whether app having access to internet (`https://github.com/react-native-community/react-native-netinfo` by using this package) or check if it able to access any local server

**Comment by SUMITIOS** [comment]
> what is it consoling ?
> check whether app having access to internet (`https://github.com/react-native-community/react-native-netinfo` by using this package) or check if it able to access any local server

The console gives an error `Network request failed` (in the catch part of the fetch promise ). It is not internet issue because at the same time, in same app all the api calls using `XMLHttpRequest` are working totally fine. I hence switched to `XMLHttp` for time being as a workaround.

**Comment by anymore1405** [comment]
You can try this https://github.com/facebook/react-native/issues/25244#issuecomment-671564438

**Comment by SUMITIOS** [comment]
@anymore1405  I have already tried that before posting flipper version =50. I will try again for a freshly created project.

**Comment by anymore1405** [comment]
> @anymore1405  I have already tried that before posting flipper version =50. I will try again for a freshly created project.

If you don't use flipper, you can try disable it in `MainApplication.java`

**Comment by SUMITIOS** [comment]
Thnx, but I use flipper, my issue seems different. is not limited to HTTP only.

**Comment by SUMITIOS** [comment]
@anymore1405 @hsnandeeshgowda   thnx both of you, but after trying every thing I else I saw some 10 npm `vulnerablities`. I ran `npm audit` and saw 6 vulnerabilities were related to `node-fetch`. I just ran `npm audit fix` , and surprisingly fetch calls  now work both in newly created and existing projects.

With this I should close the issue now, as it does not seem to be related to react native anymore.
